Store/admin/components/Account/Details.php

```diff
@@ -115,12 +115,13 @@ protected function getTableModel(SwatView $view)
 		case 'payment_methods_view':
 			return $this->getPaymentMethodsTableModel($view);
 		}
+		return null;
 	}
 
 	// }}}
 	// {{{ protected function getOrdersTableModel()
 
-	protected function getOrdersTableModel(SwatTableView $view)
+	protected function getOrdersTableModel(SwatTableView $view): SwatDBDefaultRecordsetWrapper
```

Comment:
getTableModel returns this value which is not a SwatTableModel. I think this relied on "duck" typing before where both classes are countable and iterable and those are the only public interfaces used by the TableView. I don't think PHP supports any structural typing so we would need to return an actual TableModel instead of something with the same interface. We could:

- wrap the recordset result in a tablestore, or
- relax the types

I think it's likely if we enforce a return type on getTableModel in admin we will need to make many minor updates to admin index and details pages.